### ISSUE TITLE
Feature Idea: Class as Default Export

### ISSUE THREAD

**Issue by amiller-gh** [reboundjs/rebound-api#11]
If we choose to support exporting a class as the default export, we enable two major benefits for API writers:

1. Endpoints can maintain process-dependent state on the class instance instead of in module-level variables.
2. Developers can use decorators to easily augment their endpoints with a terse syntax.

Example syntax may look like this:
```TypeScript

function basicAuth() { /* Basic HTTP Auth Impl Here */ }

export default class MyEndpoint {
  private state: any = {};

  // Anyone may GET a process dependent key value pair.
  GET(req, res){
    return { status: 'success', data: this.state[req.params.key] };
  }
 
  // Only authenticated users may set values.
  @basicAuth POST(req, res){
    this.state[req.data.key] = req.data.value; 
    return { status: 'success' };
  }

}

```

**Comment by amiller-gh** [comment]
By reserving default exports for API endpoint class constructors, and since there is no great way to detect the difference between a function and a class definition, this would require *all* default exports to be interpreted as a class constructor, breaking the current behavior of treading default exports as a GET function. I'm personally okay with that and like this API better.

**Comment by amiller-gh** [comment]
Implemented, tested, used in a product with real basic HTTP auth implemented, liked it a lot, and released in 0.0.7 🙂
